walk/walk.go

```diff
@@ -363,6 +363,9 @@ func resolveFileInfo(wc *walkConfig, dir, rel string, ent fs.DirEntry) fs.DirEnt
 type pathTrie struct {
 	children map[string]*pathTrie
 	entry    *fs.DirEntry
+
+	buildFileErr error
```

Comment:
This could probably be done different or better, but persisting this `error` here keeps the scope of this PR minimal. We can think about changing it more in the future.